### ISSUE TITLE
TS error on large nested objects

### ISSUE THREAD

**Issue by noobmaster19** [svecosystem/formsnap#169]
### Describe the bug

There's a TS error: 
```
Type instantiation is excessively deep and possibly infinite.ts(2589)
⚠ Error(TS2589) | 
Type instantiation is excessively deep and possibly infinite.
(property) form: SuperForm<{
    name: string;
    pageId: string;
    componentProperties: {
        type: "table";
        props: {
            title: string;
            schema: DTableJSONSchemaArray;
            description: string;
            enableCheckBoxes: boolean;
            ... 9 more ...;
            rowCount: number;
        };
        x: number;
        y: number;
        w: number;
        h: number;
    };
    ... 4 more ...;
    parent?: string | ... 1 more ... | undefined;
}, any>

```
On large nested objects

### Reproduction

You can reproduce by making a very large type

### System Info

```bash
System:
    OS: macOS 14.4.1
    CPU: (10) arm64 Apple M1 Pro
    Memory: 123.45 MB / 16.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 18.15.0 - ~/.nvm/versions/node/v18.15.0/bin/node
    Yarn: 1.22.19 - /opt/homebrew/bin/yarn
    npm: 9.5.0 - ~/.nvm/versions/node/v18.15.0/bin/npm
    pnpm: 8.7.5 - ~/Library/pnpm/pnpm
    bun: 1.0.14 - ~/.bun/bin/bun
  Browsers:
    Chrome: 124.0.6367.60
    Safari: 17.4.1
  npmPackages:
    @sveltejs/kit: ^2.5.6 => 2.5.6 
    formsnap: ^1.0.0 => 1.0.0 
    svelte: ^4.2.14 => 4.2.14 
    sveltekit-superforms: ^2.12.4 => 2.12.4 
    zod: ^3.22.4 => 3.22.4
```

### Severity

blocking an upgrade

**Comment by gerhardcit** [comment]
I'm curious to know how you solved this @noobmaster19? @huntabyte , I still face issues with deep objects, especially when they are optional.

Could you extend the documentation example to deal with a more compare deep object like this?

```typescript
import { z } from "zod";
export const settingsFormSchema = z.object({
	email: z.string().email(),
	bio: z.string().max(250).optional(),
	marketingEmails: z.boolean().default(true),
	language: z.enum(["en", "es", "fr"]).default("en"),
	theme: z.enum(["light", "dark"]).default("light"),
        webProps: z.object({
		enabled: z.boolean().default(true),
		color: z.string().optional(),
		fontSize: z.number().optional(),
	}).optional(),
	mobiProps: z.object({
		enabled: z.boolean().default(true),
		sizes: z.enum(["small", "medium", "large"]).default("medium"),
	}).optional(),
});
```

binding forms to form.webProps.color becomes messy when fields are optional.

It feels like forms are really only designed for 1 level objects?

(I must add that I got spoiled from wayback with angularjs days where ng-bind="object.subobject.options.color" was very basic.. and if you fill it in, I think objectPath.set took care of making sure the object exists..
I know, not very strong types, but still. I keep running in circles when binding to nested forms with optional fields.

I would wish to know how you deal with that?

It also gets really messy when I derive types from inferred zod types based on "json" fields stored as text.

```typescript
import { sqliteTable, text } from "drizzle-orm/sqlite-core";
import { createInsertSchema } from "drizzle-zod";
import { z } from 'zod';

type TenantMeta = {
    clientName?: string;
    mainColor?: string;
    accessCode?: string;
}

export const tenants = sqliteTable('tenants', {
    id: text('id').primaryKey(),
    apiUrl: text('api_url').notNull(),
    imageUrl: text('image_url').notNull(),
    meta: text('meta', { mode: 'json' }).notNull().$type<TenantMeta>(),
})

export type TTenant = typeof tenants.$inferSelect
// super forms does not understand this well, and binding to the form is a bit tricky
export const tenantInsertSchema = createInsertSchema(tenants);

// for superforms to work, you have to be more explicit
export const tenantFormSchema = createInsertSchema(tenants, {
    meta: z.object({
        clientName: z.string().optional(),
        mainColor: z.string().optional(),
        accessCode: z.string().optional(),
    }).required()
});

// use this type when you need to refer to the form
export type TTenantForm = typeof tenantFormSchema;
```

later when I do this in +page.server.ts
```
// cloudflare binding
const db = drizzle(platform!.env.D1)

 const dbTenant = await db.select()
            .from(tenants)
            .where(eq(tenants.id, id))
            .get();

if (!dbTenant) {
    return error(404, 'Not found');
}

// found the user, validate it and return it in the form
 const form = await superValidate(dbTenant, zod(tenantFormSchema));
```

<img width="1024" alt="image" src="https://github.com/user-attachments/assets/d7df5ff8-ab5f-4a84-bb65-0061a7603ab1">

**Comment by huntabyte** [comment]
This should perhaps be brought to the attention of `sveltekit-superforms` and not formsnap.

**Comment by gerhardcit** [comment]
@huntabyte , thanks, yes, now that I look at it closely, that is correct.
However, after spending too much time trying to handle nested object, I changed strategy to create DTO between DB json and form json. This way I have full control over what goes to the form and I can test mappings properly.
So far it its working a lot better if you have form data on the root level.

I got sucked to this level starting at https://next.shadcn-svelte.com/examples/tasks which is great, 
worked my way down from there..  after 15 years of doing bootstrap... :)
